Don't re-compile org-odt.el at install time.
Patch submitted upstream:
http://lists.gnu.org/archive/html/emacs-orgmode/2012-01/msg00005.html

--- org-7.8.03-orig/Makefile
+++ org-7.8.03/Makefile
@@ -247,14 +247,10 @@
 g:
 	${MAKE} pdf && open doc/orgguide.pdf
 
-# Always force re-compilation of org-odt
-lisp/org-odt.elc: org-odt-data-dir
-org-odt-data-dir:
-
 # Sleight of hand to "hard code" the value of $(datadir) in
 # org-odt.el.  See variables `org-odt-styles-dir-list' and
 # `org-odt-schema-dir-list'.
-install-lisp: BATCH_EXTRA = -eval "(setq org-odt-data-dir (expand-file-name \"$(datadir)\"))"
+lisp/org-odt.elc: BATCH_EXTRA = -eval "(setq org-odt-data-dir (expand-file-name \"$(datadir)\"))"
 
 install-lisp: $(LISPFILES) $(ELCFILES)
 	if [ ! -d $(lispdir) ]; then $(MKDIR) $(lispdir); else true; fi ;
